homeassistant/config.py

```diff
@@ -1025,21 +1045,175 @@ async def merge_packages_config(
+async def async_process_component_and_handle_errors(
+    hass: HomeAssistant,
+    config: ConfigType,
+    integration: Integration,
+    raise_on_failure: bool = False,
 ) -> ConfigType | None:
-    """Check component configuration and return processed configuration.
+    """Process and component configuration and handle errors.
+
+    In case of errors:
+    - Print the error messages to the log.
+    - Raise a ConfigValidationError if raise_on_failure is set.
+
+    Returns the integration config or `None`.
+    """
+
+    def _get_log_message_and_stack_print_pref(
+        translation_key: str, ex: Exception, domain: str, platform_name: str
+    ) -> tuple[str | None, bool]:
+        """Get message to log and print stack trace preference."""
+        log_message_mapping: dict[str, tuple[str, bool]] = {
+            "component_import_err": (f"Unable to import {domain}: {ex}", False),
+            "config_platform_import_err": (
+                f"Error importing config platform {domain}: {ex}",
+                False,
+            ),
+            "config_validator_unknown_err": (
+                f"Unknown error calling {domain} config validator",
+                True,
+            ),
+            "config_schema_unknown_err": (
+                f"Unknown error calling {domain} CONFIG_SCHEMA",
+                True,
+            ),
+            "platform_validator_unknown_err": (
+                f"Unknown error validating {platform_name} platform config with {domain} "
+                "component platform schema",
+                True,
+            ),
+            "platform_component_load_err": (f"Platform error: {domain} - {ex}", False),
+            "platform_component_load_exc": (f"Platform error: {domain} - {ex}", True),
+            "platform_schema_validator_err": (
+                f"Unknown error validating config for {platform_name} platform "
+                f"for {domain} component with PLATFORM_SCHEMA",
+                True,
+            ),
+        }
+        return log_message_mapping.get(translation_key, (None, False))
+
+    integration_config_info = await async_process_component_config(
+        hass, config, integration
+    )
+    processed_config = integration_config_info.config
+    config_exception_info = integration_config_info.exception_info_list
+    if not config_exception_info:
+        return processed_config
+
+    platform_exception: ConfigExceptionInfo
+    config_error_messages: list[
+        tuple[str, ConfigExceptionInfo, str, str, int | str]
+    ] = []
+    general_error_messages: list[tuple[str, ConfigExceptionInfo]] = []
+    domain = integration.domain
+    for platform_exception in config_exception_info:
+        link = platform_exception.integration_link or integration.documentation
+        ex = platform_exception.exception
+        p_name = platform_exception.platform_name
+        p_config = platform_exception.config
+        config_file: str = "?"
+        line: int | str = "?"
```

Comment:
All these variables should be replaced with a dataclass instance

Reply:
Not needed anymore